ls1intum/Artemis
# `Exam mode`: Fix breadcrumbs in test exams

### Checklist
#### General
#### Client

### Motivation and Context

Fixes issue no. 3, no. 21 and no.24 found during [exam mode testing](https://confluence.ase.in.tum.de/pages/viewpage.action?pageId=138610767).

### Description

Changes the navbar breadcrumbs to translate and show 'TestExam' with the right url for issue 3 and 21.
Bugfix to correctly set the testExam property in the exam-participation component instead of relying on the url params for issue 24.
Change the url and corresponding breadcrumb when starting a test exam from 'new' to 'start'

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students

1. Log in to Artemis
2. Create Test Exam
3. Participate in the test exam and make sure the issues described don't happen anymore
4. Make sure participations in test runs and in non-test exams are unaffected

### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|
| exam-participation.component.ts | 65% | 76% |

### Screenshots

Changed url and breadcrumbs

#### Before

![image](https://user-images.githubusercontent.com/11006274/196530028-b418aedd-a6c4-4f82-884c-2c3f95e39657.png)

#### After

![startBreadCrumb](https://user-images.githubusercontent.com/11006274/196530063-e1cf2338-3fdc-4afa-87fb-a03faf465cad.png)
